### PR TITLE
HPCC-25837 Remove ESP ws_machine service from container build

### PR DESCRIPTION
**Pull request by wangkx**
The ws_machine service is only used for bare metal environment.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by wangkx** [comment]
@ghalliday please review. If you want, I can wait until after https://github.com/hpcc-systems/HPCC-Platform/pull/14917 is merged.

**Review by ghalliday — COMMENTED** [review]
This should target master rather than 8.0.x.

Also, shouldn't there be some other related changes - e.g. application.yaml, eclwatch.yaml

@GordonSmith  Are there any dependencies of eclwatch js/ts code on any of the functions in the service?

**Comment by wangkx** [comment]
I will re-target it to master.
I do find some related changes in the yaml files and will fix them.
I also find ws_machine from the esp.xml inside dockerfiles/platform-core and dockerfiles/platform-core-debug. @jakesmith do we still need those esp xml files? Should I remove the ws_machine lines from those esp xml files?

**Comment by richardkchapman** [comment]
AFAIK the esp.xml inside the Dockerfiles should not be there any more and should not be in use. @afishbeck Do you know otherwise?

**Comment by ghalliday** [comment]
@wangkx please can you describe how this has been tested

File: esp/scm/espscm.cmake [hunk 1]
```diff
@@ -23,6 +23,35 @@
 set ( ESPSCM_SOURCE_DIR ${HPCC_SOURCE_DIR}/esp/scm )
 set ( ESPSCM_GENERATED_DIR ${CMAKE_BINARY_DIR}/generated )
 
+#ifdef _CONTAINERIZED
```
Comment (ghalliday) [inline esp/scm/espscm.cmake:26]: This is not how conditions are added to cmake files - the #s will be treated as comments.

**Comment by afishbeck** [comment]
The esp.xml file is not used by k8s ESPs so should be able to be removed.

**Comment by wangkx** [comment]
@ghalliday I am going to do several tests. 1. in bare metal environment. 1.1 add some bad code to see if the #ifdef works (should see an error). 1.2 After the build finishes correctly, run the build. 1.3 in the activity page, check the disk usage section. In System server page, check the preflight function. 2. in cloud environment, do the similar as 1.1 and 1.2.

File: esp/scm/espscm.cmake (hunk 1 above)
Comment (wangkx) [reply esp/scm/espscm.cmake:26]: Sorry, my mistake

**Comment by wangkx** [comment]
@afishbeck @ghalliday please review

File: esp/scm/espscm.cmake [hunk 2]
```diff
@@ -23,6 +23,35 @@
 set ( ESPSCM_SOURCE_DIR ${HPCC_SOURCE_DIR}/esp/scm )
 set ( ESPSCM_GENERATED_DIR ${CMAKE_BINARY_DIR}/generated )
 
+if (CONTAINERIZED)
```
[the remark this answers is not in the document]
Comment (afishbeck) [reply esp/scm/espscm.cmake:26]: Just fixed the variable name in my example.

[the remark this answers is not in the document]
Comment (wangkx) [reply esp/scm/espscm.cmake:26]: Thanks for letting me know. I will fix.

**Comment by wangkx** [comment]
Duplicated code is removed. The PR is re-targeted to the master.

**Review by afishbeck — APPROVED** [review]
@wangkx looks good.

**Review by ghalliday — APPROVED** [review]
Now looks ready.  I will merge before my change.

**Pull request merged**